clients/fides-js/src/components/notices/NoticeOverlay.tsx

```diff
@@ -309,6 +330,30 @@ const NoticeOverlay: FunctionComponent<OverlayProps> = ({
     parsedCookie?.consent,
   ]);
 
+  const handleToggleChange = useCallback(
+    (updatedKeys: Array<string>, preference: FidesEventDetailsPreference) => {
+      const eventExtraDetails: FidesEvent["detail"]["extraDetails"] = {
+        servingComponent: servingComponentRef.current as NonNullable<
+          FidesEvent["detail"]["extraDetails"]
+        >["servingComponent"],
+        trigger: triggerRef.current,
+        preference,
+      };
```

Comment:
This seems pretty verbose, mostly due to types I guess. Can you think of any way to simplify constructing `eventExtraDetails` at all?

Reply:
Agreed. This is one of the unfortunate side effects of using Docs also as our dev Types. I can create an alias for this though.